tests/Server/TextDocument/CompletionTest.php

```diff
@@ -716,6 +716,28 @@ public function testMethodReturnType()
         ], true), $items);
     }
 
+    public function testMethodReturnSelf()
```

Comment:
I find it a bit weird that these are tests of `textDocument/completion`. Clearly what's being tested here is the DefinitionResolver?

Reply:
Makes sense, I've removed these tests and the cases are now covered with the validation cases. Is that what you meant?

Reply:
I was thinking of a unit test for DefinitionResolver, but this seems sufficient!